src/Microsoft.ML.Core/Data/ReadOnlyMemoryUtils.cs

```diff
@@ -0,0 +1,285 @@
+            if (span[ichMin] != ' ' && span[ichLim - 1] != ' ')
+                return memory;
+
+            while (ichMin < ichLim && span[ichMin] == ' ')
+                ichMin++;
+            while (ichMin < ichLim && span[ichLim - 1] == ' ')
+                ichLim--;
+            return memory.Slice(ichMin, ichLim - ichMin);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> with leading and trailing whitespace trimmed.
+        /// </summary>
+        public static ReadOnlyMemory<char> TrimWhiteSpace(ReadOnlyMemory<char> memory)
+        {
+            if (memory.IsEmpty)
+                return memory;
+
+            int ichMin = 0;
+            int ichLim = memory.Length;
+            var span = memory.Span;
+            if (!char.IsWhiteSpace(span[ichMin]) && !char.IsWhiteSpace(span[ichLim - 1]))
+                return memory;
+
+            while (ichMin < ichLim && char.IsWhiteSpace(span[ichMin]))
+                ichMin++;
+            while (ichMin < ichLim && char.IsWhiteSpace(span[ichLim - 1]))
+                ichLim--;
+
+            return memory.Slice(ichMin, ichLim - ichMin);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> with trailing whitespace trimmed.
+        /// </summary>
+        public static ReadOnlyMemory<char> TrimEndWhiteSpace(ReadOnlyMemory<char> memory)
+        {
+            if (memory.IsEmpty)
+                return memory;
+
+            int ichLim = memory.Length;
+            var span = memory.Span;
+            if (!char.IsWhiteSpace(span[ichLim - 1]))
+                return memory;
+
+            while (0 < ichLim && char.IsWhiteSpace(span[ichLim - 1]))
+                ichLim--;
+
+            return memory.Slice(0, ichLim);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> with trailing whitespace trimmed.
+        /// </summary>
+        public static ReadOnlyMemory<char> TrimEndWhiteSpace(ReadOnlyMemory<char> memory, ReadOnlySpan<char> span)
+        {
+            if (memory.IsEmpty)
+                return memory;
+
+            int ichLim = memory.Length;
+            if (!char.IsWhiteSpace(span[ichLim - 1]))
+                return memory;
+
+            while (0 < ichLim && char.IsWhiteSpace(span[ichLim - 1]))
+                ichLim--;
+
+            return memory.Slice(0, ichLim);
+        }
+
+        public static NormStr AddToPool(ReadOnlyMemory<char> memory, NormStr.Pool pool)
+        {
+            Contracts.CheckValue(pool, nameof(pool));
+            return pool.Add(memory);
+        }
+
+        public static NormStr FindInPool(ReadOnlyMemory<char> memory, NormStr.Pool pool)
+        {
+            Contracts.CheckValue(pool, nameof(pool));
+            return pool.Get(memory);
+        }
+
+        public static void AddToStringBuilder(ReadOnlyMemory<char> memory, StringBuilder sb)
```

Comment:
I wonder if we can get rid of this method and just have all the callers call `sb.Append(memory)` themselves. That way we don't have yet another public util method that does the same thing.